### PR TITLE
chore: use `composite` builds for tests

### PR DESCRIPTION
**Pull request by jrandolf-google**
This PR does a few things:

- [MAIN] Migrates all building to use TypeScript's `composite` compilation. See https://www.typescriptlang.org/docs/handbook/project-references.html. As a result:
  - The test server is now programmed in TypeScript.
  - The tests are now in a `src` file.
  - `ts-node` was removed for tests and, instead, tests are directly compiled before testing.
- `eslint-formatter-codeframe` was added due to warnings by ESLint.
- `coverage-utils.js` has been removed.
- Some unnecessary files have been removed from `.gitignore`, `.eslintignore`, and `.prettierignore`.

### CODE REVIEW

**Comment by payephyoeaung87** [comment]
gh repo clone puppeteer/puppeteer

**Comment by payephyoeaung87** [comment]
chore: use composite builds for tests

**Review by payephyoeaung87 — COMMENTED** [review]
feat: adds v4 UUID to crypto

This adds support for v4 UUIDs to the library.

fix(utils): unicode no longer throws exception
  PiperOrigin-RevId: 345559154
  BREAKING-CHANGE: encode method no longer throws.
  Source-Link: googleapis/googleapis@5e0dcb2

feat(utils): update encode to support unicode
  PiperOrigin-RevId: 345559182
  Source-Link: googleapis/googleapis@e5eef86
 "adds v4 UUID to crypto"
 "unicode no longer throws exception"
 "update encode to support unicode".

**Comment by payephyoeaung87** [comment]
If my additional letters are true or convenient, can I get suggestions?
Please advise me

၂၀၂၂၊ ဇွန် ၁၅၊ ဗုဒ္ဓဟူး နံနက် ၂:၁၉ တွင် Pyae Phyoe Aung <
***@***.***> က ရေးခဲ့သည်−

**Comment by payephyoeaung87** [comment]
If my additional letters are true or convenient, can I get suggestions? Please advise me

**Comment by payephyoeaung87** [comment]
git clone https://github.com/puppeteer/puppeteer
cd puppeteer

**Comment by payephyoeaung87** [comment]
npm install

**Comment by payephyoeaung87** [comment]
npm run test:unit

File: package.json [hunk 1]
```diff
@@ -27,15 +27,15 @@
...
+    "test": "npm run lint --silent && npm run test:unit:coverage",
+    "test:unit": "npm run build:test && mocha",
     "test:unit:firefox": "cross-env PUPPETEER_PRODUCT=firefox npm run test:unit",
     "test:unit:coverage": "c8 --check-coverage --lines 94 npm run test:unit",
     "test:unit:chrome-headless": "cross-env HEADLESS=chrome npm run test:unit",
     "test:protocol-revision": "ts-node -s scripts/ensure-correct-devtools-protocol-package",
     "test:pinned-deps": "ts-node -s scripts/ensure-pinned-deps",
     "test:install": "scripts/test-install.sh",
-    "test:debug": "npm run build:tsc:cjs && npm run build:tsc:cjs:compat && mocha --inspect-brk --config mocha-config/puppeteer-unit-tests.js",
+    "test:debug": "npm run build:test mocha --inspect-brk",
```
Comment (OrKoN) [inline package.json]: `test && mocha`?

**Review by OrKoN — APPROVED** [review]

**Pull request merged**